### PR TITLE
Linux support

### PR DESCRIPTION
**Pull request by Yohello1**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ if (CMAKE_CXX_SIMULATE_ID STREQUAL "MSVC" AND CMAKE_CXX_COMPILER_ID STREQUAL "Cl
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /GR- /arch:AVX /MT /W4 /EHa /EHs")
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     message(STATUS "Using Clang/GCC compiler options")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native -fpermissive")
```
Comment (LegendWasTaken) [inline CMakeLists.txt:14]: Should we add `-Wall` `-Werror` `-pedantic` for more warnings / catching things? (And the same on MSVC)

File: CMakeLists.txt [hunk 2]
```diff
@@ -61,11 +61,20 @@ add_executable(CRender src/main.cpp
...
+        src/render/entities/registry.h 
+        src/render/draft/draft_renderer.cpp 
+        src/render/draft/draft_renderer.h 
+        src/render/timer.cpp 
+        src/render/timer.h 
+        src/util/logger.h 
+        src/util/logger.cpp 
+        src/util/numbers.h 
+        src/render/brdf.h 
+        src/util/denoise.h)
```
Comment (LegendWasTaken) [inline CMakeLists.txt:73]: Thank you for doing this

File: CMakeLists.txt [hunk 3]
```diff
@@ -61,11 +61,20 @@ add_executable(CRender src/main.cpp
...
 
 target_include_directories(CRender PRIVATE src)
 target_include_directories(CRender PRIVATE external)
 
-target_link_libraries(CRender glfw fmt glm embree OpenImageDenoise)
+target_link_libraries(CRender glfw fmt glm embree OpenImageDenoise stdc++fs)
```
Comment (LegendWasTaken) [inline CMakeLists.txt:78]: I don't think we need to do this. (This seems to be needed to be done when there's an error somewhere else)

File: CMakeLists.txt [hunk 4]
```diff
@@ -61,11 +61,20 @@ add_executable(CRender src/main.cpp
...
 
-target_compile_definitions(CRender PUBLIC -DGLFW_INCLUDE_NONE -D__STDC_CONSTANT_MACROS)
+target_compile_definitions(CRender PUBLIC -DGLFW_INCLUDE_NONE -D__STDC_CONSTANT_MACROS -DIMGUI_IMPL_OPENGL_LOADER_GLAD)
```
Comment (LegendWasTaken) [inline CMakeLists.txt:80]: No new line EOF

File: readme.md [hunk 5]
```diff
@@ -28,6 +28,23 @@ If you're a developer, and want to get started and [contribute](#Contributing) t
...
 
 ***
-## Contributing
+## Building
+
+### Linux
+
+1) To build this project you will require [Intel Rendering Toolkit](https://software.intel.com/content/www/us/en/develop/tools/oneapi/rendering-toolkit/download.html) (online version), CMake, and `libxi-dev build-essential xorg libglfw3 libglfw3-dev libxinerama-dev libxcursor-dev` (these are the packages for Debian/Ubuntu)
+
+2) You will need to run `source /opt/intel/oneapi/setvars.sh` so it sets the global vars needed for building. If this doesn't work please look for this file, and source that file in the same terminal window you compile in.
```
Comment (LegendWasTaken) [inline readme.md:37]: You also need onetbb from Intel's multi-threading kit.

File: readme.md [hunk 6]
```diff
@@ -28,6 +28,23 @@ If you're a developer, and want to get started and [contribute](#Contributing) t
 5) Run `CRender.exe` and enjoy!
 
 ***
-## Contributing
+## Building
+
+### Linux
+
+1) To build this project you will require [Intel Rendering Toolkit](https://software.intel.com/content/www/us/en/develop/tools/oneapi/rendering-toolkit/download.html) (online version), CMake, and `libxi-dev build-essential xorg libglfw3 libglfw3-dev libxinerama-dev libxcursor-dev` (these are the packages for Debian/Ubuntu)
```
Comment (LegendWasTaken) [inline readme.md:35]: Specify you only need Embree 3.0 and OpenImageDenoise?

File: readme.md [hunk 7]
```diff
@@ -28,6 +28,23 @@ If you're a developer, and want to get started and [contribute](#Contributing) t
...
 
+#### Bugs/issues with building:
+If you get an error such as `./CRender: symbol lookup error: /opt/intel/oneapi/oidn/1.4.0/lib/libOpenImageDenoise.so.1: undefined symbol: _ZN3tbb6detail2r15spawnERNS0_2d14taskERNS2_18task_group_contextE` you need to remove every tbb package except the intel one.
+If if you get a `glenable` etc error you need to get new drivers
+If it cannot find embree you will need to go to `/opt/intel/oneapi` and move embree to a folder called `embree` instead of `embreeX.XXX`
+
+***
+## Contributing
 If you would like to contribute to this repository, just make a half decent PR. I have no requirements for now...
+OB[3~[3~
```
Comment (LegendWasTaken) [inline readme.md:50]: ? Encoding issues?

**Review by LegendWasTaken — CHANGES_REQUESTED** [review]
Thanks for the work, some things to be talked about before merging.

File: readme.md (hunk 7 above)
Comment (Yohello1) [reply readme.md:50]: Yeah... Didn't notice it

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Yohello1) [reply CMakeLists.txt:14]: I changed up the code to just rename the shade thingy from `type` to `shade_type` so it stops throwing the error.

Comment (Yohello1) [reply CMakeLists.txt:14]: `-Werror` just breaks everything for no good reason, like the random funcs. Then `-pedantic` I don't see why we would need this flag. `-Wall` gives us warnings which we can/have to ignore, such as it saying `comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’`. I personally don't see why we would need any of these.

File: readme.md (hunk 6 above)
Comment (Yohello1) [reply readme.md:35]: Those have been added in.

File: readme.md (hunk 5 above)
Comment (Yohello1) [reply readme.md:37]: Yup, I added those in withing the latest commit.

File: CMakeLists.txt (hunk 4 above)
Comment (Yohello1) [reply CMakeLists.txt]: So just add in a new line?

File: CMakeLists.txt (hunk 3 above)
Comment (Yohello1) [reply CMakeLists.txt]: Unless someone is using a compiler from the 1800's (old one) we won't need it, might as well remove it then.

File: CMakeLists.txt (hunk 1 above)
Comment (LegendWasTaken) [reply CMakeLists.txt]: Its good to have these, but yeah maybe not in this pr. Ill do it myself at some point.

File: CMakeLists.txt (hunk 4 above)
Comment (LegendWasTaken) [reply CMakeLists.txt]: Yeah

Comment (LegendWasTaken) [reply CMakeLists.txt]: Actually nah its fine, its the cmakeliste file

**Comment by LegendWasTaken** [comment]
Looks good to me :)

**Pull request merged**